app/src/main/java/com/github/ybecker/epforuml/SavedQuestionsFragment.kt

```diff
@@ -13,13 +20,59 @@ import android.view.ViewGroup
  * create an instance of this fragment.
  */
 class SavedQuestionsFragment : Fragment() {
+    private lateinit var recyclerView: RecyclerView
+    private lateinit var user : Model.User
+
+    private lateinit var cache : ArrayList<Model.Question>
+    private lateinit var newIntentDetails : Intent
+
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        // Inflate the layout for this fragment
+        cache = this.requireArguments().getParcelableArrayList("savedQuestions")!!
 
+        // Inflate the layout for this fragment
         return inflater.inflate(R.layout.fragment_saved_questions, container, false)
+    }
+
+    @SuppressLint("SetTextI18n")
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        // initialize recycler with saved questions
+        user = DatabaseManager.user ?: Model.User()
+        val userId = user.userId
+
+        if (userId.isNotEmpty() && cache.isNotEmpty()) {
+            val layoutManager = LinearLayoutManager(context)
+            recyclerView = view.findViewById(R.id.recycler_saved_questions)
+            recyclerView.layoutManager = layoutManager
+
+            // display saved questions
+            val adapter = ForumAdapter(cache)
```

Comment:
Here you should not use the ForumAdapter as it require a connection to use firebase, you should create an adapter that work only with the cache. 
Note that opening the questionDetails don't crash but the answer cannot load without internet connection, so you will need to cache answers as well.